core/templates/dev/head/domain/objects/NumberWithUnitsObjectFactory.js

```diff
@@ -119,23 +147,61 @@ oppia.factory('NumberWithUnitsObjectFactory', [
             value = rawInput.substr(0, ind).trim();
             units = rawInput.substr(ind).trim();
           }
+
+          var keys = Object.keys(CURRENCY_UNITS);
+          for (var i = 0; i < keys.length; i++) {
+            for (var j = 0;
+              j < CURRENCY_UNITS[keys[i]].front_units.length; j++) {
+              if (units.indexOf(
+                CURRENCY_UNITS[keys[i]].front_units[j]) !== -1) {
+                throw new Error(
+                  NUMBER_WITH_UNITS_PARSING_ERRORS.INVALID_CURRENCY_FORMAT);
+              }
+            }
+          }
         } else {
-          if (!rawInput.startsWith('$ ') && !rawInput.startsWith('Rs ') &&
-            !rawInput.startsWith('₹ ')) {
+          var startsWithCurrencyUnits = false;
+          var keys = Object.keys(CURRENCY_UNITS);
+          for (var i = 0; i < keys.length; i++) {
+            for (var j = 0;
+              j < CURRENCY_UNITS[keys[i]].front_units.length; j++) {
+              if (rawInput.startsWith(CURRENCY_UNITS[keys[i]].front_units[j])) {
+                startsWithCurrencyUnits = true;
+                break;
+              }
+            }
+          }
+          if (startsWithCurrencyUnits === false) {
             throw new Error(NUMBER_WITH_UNITS_PARSING_ERRORS.INVALID_CURRENCY);
           }
-          var ind = rawInput.indexOf(' ');
+          var ind = rawInput.indexOf(rawInput.match(/[0-9]/));
           if (ind === -1) {
             throw new Error(NUMBER_WITH_UNITS_PARSING_ERRORS.INVALID_CURRENCY);
           }
-          units = rawInput.substr(0, ind) + ' ';
+          units = rawInput.substr(0, ind).trim() + ' ';
```

Comment:
Consider appending space after the loop is finished below. So that condition inside loop doesn't need `+ ' '`.

Reply:
Done